specification/appplatform/resource-manager/Microsoft.AppPlatform/preview/2021-09-01-preview/appplatform.json

```diff
@@ -3715,6 +3927,156 @@
         }
       }
     },
+    "StorageResource": {
+      "description": "Storage resource payload.",
+      "type": "object",
+      "allOf": [
+        {
+          "$ref": "#/definitions/ProxyResource"
+        }
+      ],
+      "properties": {
+        "properties": {
+          "$ref": "#/definitions/StorageProperties",
+          "description": "Properties of the storage resource payload.",
+          "x-ms-client-flatten": false
+        }
+      }
+    },
+    "StorageProperties": {
+      "description": "Storage resource payload.",
+      "type": "object",
+      "properties": {
+        "storageType": {
+          "description": "The type of the storage.",
+          "type": "string"
+        }
+      },
+      "discriminator": "storageType"
+    },
+    "StorageAccount": {
+      "description": "storage resource of type Azure Storage Account.",
+      "type": "object",
+      "required": [
+        "storageType",
+        "accountName",
+        "accountKey"
+      ],
+      "allOf": [
+        {
+          "$ref": "#/definitions/StorageProperties"
+        }
+      ],
+      "properties": {
+        "accountName": {
+          "description": "The account name of the Azure Storage Account.",
+          "type": "string"
+        },
+        "accountKey": {
+          "description": "The account key of the Azure Storage Account.",
+          "type": "string"
+        }
+      },
+      "x-ms-discriminator-value": "StorageAccount"
+    },
+    "StorageResourceCollection": {
+      "description": "Collection compose of storage resources list and a possible link for next page.",
+      "type": "object",
+      "properties": {
+        "value": {
+          "description": "The storage resources list.",
+          "uniqueItems": false,
+          "type": "array",
+          "items": {
+            "$ref": "#/definitions/StorageResource"
+          }
+        },
+        "nextLink": {
+          "description": "The link to next page of storage list.",
+          "type": "string"
+        }
+      }
+    },
+    "CustomPersistentDiskResource": {
+      "description": "Custom persistent disk resource payload.",
+      "type": "object",
+      "required": [
+        "storageId"
+      ],
+      "properties": {
+        "properties": {
+          "$ref": "#/definitions/CustomPersistentDiskProperties",
+          "description": "Properties of the custom persistent disk resource payload.",
+          "x-ms-client-flatten": false
+        },
+        "storageId": {
```

Comment:
In this https://github.com/Caoxuyang/azure-rest-api-specs/pull/1/files, the similar property is called "resourceId", shall we make such similar properties consistent? e.g. here can be "storageResourceId", etc. You can talk w/ Jie and make you guys payload consistent.